src/client/clientmap.cpp

```diff
@@ -433,13 +444,43 @@ void ClientMap::renderMap(video::IVideoDriver* driver, s32 pass)
 			}
 		}
 	}
+
+	// Render all transparent layers in order
```

Comment:
Can we avoid the duplication of the code?

Reply:
Yes, I've changed it so MeshBufListList is only used to generate draw_order list. As a bonus, setMaterial is now only called when really necessary.